python/pyspark/core/context.py

```diff
@@ -653,7 +653,7 @@ def stop(self) -> None:
         if getattr(self, "_jsc", None):
             try:
                 self._jsc.stop()
-            except Py4JError:
+            except (Py4JError, ConnectionRefusedError):
```

Comment:
Is this considered a behavior change? Regardless of whether this error is caught, it can still reach the `finally` block. I think it's better to throw the original exception than to give a warning message.